frontend/src/components/Player.tsx

```diff
@@ -8,17 +8,38 @@ declare global {
 }
 
 class Player extends React.Component<YT.PlayerOptions> {
+  private internalPlayer: any;
+
   constructor(props: YT.PlayerOptions) {
     super(props);
     const tag = document.createElement("script");
     tag.src = "https://www.youtube.com/iframe_api";
     const firstScriptTag = document.getElementsByTagName("script")[0];
     firstScriptTag?.parentNode?.insertBefore(tag, firstScriptTag);
     window["onYouTubeIframeAPIReady"] = () => {
-      new window["YT"].Player("player", this.props);
+      this.internalPlayer = new window["YT"].Player("player", this.props);
     };
   }
 
+  shouldUpdateVideo(prevProps: YT.PlayerOptions, props: YT.PlayerOptions) {
+    return prevProps.videoId !== props.videoId;
+  }
+
+  updateVideo(): void {
+    if (typeof this.props.videoId === "undefined" || this.props.videoId === null) {
```

Comment:
the condition can be simplified to `!this.props.videoId`

Reply:
Fixed!